app/contacts/contact_tcp.py

```diff
@@ -73,18 +77,19 @@ async def accept(self, reader, writer):
         agent, _ = await self.services.get('contact_svc').handle_heartbeat(**profile)
         new_session = Session(id=self.generate_number(size=6), paw=agent.paw, connection=connection)
         self.sessions.append(new_session)
-        await self.send(new_session.id, agent.paw)
+        await self.send(new_session.id, agent.paw, timeout=5)
 
-    async def send(self, session_id, cmd):
+    async def send(self, session_id, cmd, timeout=60):
         try:
             conn = next(i.connection for i in self.sessions if i.id == int(session_id))
             conn.send(str.encode(' '))
             conn.send(str.encode('%s\n' % cmd))
-            response = await self._attempt_connection(conn, 3)
+            response = await self._attempt_connection(session_id, conn, timeout=timeout)
             response = json.loads(response)
             return response['status'], response['pwd'], response['response'], response.get('agent_reported_time', None)
         except Exception as e:
-            return 1, '~$ ', e
+            self.log.exception(e)
+            return 1, '~$ ', e, None
```

Comment:
good catch here, maybe a good case of why typing would be nice..

Reply:
You mean stating the return value types in the function signature? That would definitely be nice to have all around

Reply:
yea, APIv2 new code is doing it, but it would have been easy to catch the fact 3 values wont unpack to 4 for the calling function or that the return value was off last time someone changed it.
I will have to double check but it may be something we are moving to implement.

Reply:
I added it for this function anyway just so it's clearer